modules/payments/backend/modules/payments/viewsets.py

```diff
@@ -60,3 +69,41 @@ def get(self, request, *args, **kwargs):
             "data": history
         }
         return Response(response, status=status.HTTP_200_OK)
+
+class AppleIAProductsView(APIView):
+    authentication_classes = [authentication.TokenAuthentication]
+    permission_classes = [permissions.IsAuthenticated]
+
+    def get(self, request, *args, **kwargs):
+        products = AppleIAPProduct.objects.filter(is_active=True)
+        response = {
+            "success": True,
+            "data": [obj.as_dict() for obj in products]
+        }
+        return Response(response, status=status.HTTP_200_OK)
+
+
+class AppleIAPayment(ViewSet):
+    serializer_class = appleIAPSerializer
+    authentication_classes = [authentication.TokenAuthentication]
+    permission_classes = [permissions.IsAuthenticated]
+
+    @csrf_exempt
+    def create(self, request):
+        data_copy = request.data.copy()
+        data_copy['user'] = request.user
+        serializer = self.serializer_class(
+            data=data_copy, context={"request": request}
+        )
+        data = None
+        if serializer.is_valid(raise_exception=True):
+            verify_receipt, success = ApplePaymentService.verify_apple_receipt(request.data)
+            print('verify_receipt', verify_receipt)
+            if success:
+                data = "success"
+            else:
+                data = "fail"
```

Comment:
Should the failure case be a different response status code? Perhaps you could do one `Response` for each case here.

Reply:
True, this needs to be documented as well because this is the place where the dev who is using this module will put his/her logic. like if receipt passes the test change his profile user_type to pro or whatever the project needs etc.

Reply:
Great, yeah, that makes sense. We should add that information in the README.md

Reply:
I have mixed feelings at best about the encapsulating the true status of the request inside the payload instead of using the response status - as it is intended in REST with HTTP verbs - but regardless of that, it should also include the logic in the source code for the dev